tests/eas/misfit.py

```diff
@@ -0,0 +1,400 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2018, Arm Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import json
+import numpy as np
+import pandas as pd
+
+from collections import OrderedDict
+from copy import deepcopy
+from unittest import SkipTest
+
+from test import LisaTest, experiment_test
+from trace import Trace
+
+from devlib.utils.misc import memoized
+
+WORKLOAD_PERIOD_MS =  16
+
+SD_ASYM_CPUCAPACITY = 0x0040
+
+class _MisfitMigrationBase(LisaTest):
+    """
+    Base class for shared functionality of misfit migration tests
+    """
+
+    test_conf = {
+        "ftrace" : {
+            "events" : [
+                "sched_switch",
+                "sched_wakeup",
+                "sched_wakeup_new",
+                "cpu_idle",
+            ],
+            "buffsize" : 100 * 1024
+        },
+        "modules": ["cgroups", "cpufreq"],
+    }
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_MisfitMigrationBase, cls).runExperiments(*args, **kwargs)
+
+    @memoized
+    @staticmethod
+    def _get_cpu_capacity(test_env, cpu):
+        """
+        :returns: The capacity value of a given CPU on the target
+        """
+        if test_env.nrg_model:
+            return test_env.nrg_model.get_cpu_capacity(cpu)
+
+        return test_env.target.read_int(
+            '/sys/devices/system/cpu/cpu{}/cpu_capacity'.format(cpu))
```

Comment:
This sound like an API suitable for devlib... isn't it?

```python
test_env.target.cpu_capacity(cpu)
```

Reply:
AFAIR the sysfs entry isn't present for older kernels, so kind of rely on the energy model values, and those are known to LISA only. It could be possible to squash that into devlib but it'd be tedious.